File: eTierMeeting_v2/API/DTO/DeliveryDTO.cs
using System;

namespace eTierV2_API.DTO
{
    public class DeliveryDTO
    {
        public string Comfirmed_Date { get; set; }
        public string Plan_Finish_Date { get; set; }
        public string Plan_Ship_Date { get; set; }
        public DateTime? Plan_Ship_Date_N { get; set; }
        public string Line_ID_ASY { get; set; }
        public string Line_ID_STC { get; set; }
        public string MO_No { get; set; }
        public string MO_Seq { get; set; }
        public string Model_Name { get; set; }
        public string Article { get; set; }
        public decimal? Plan_Qty { get; set; }
        public int UTN_Yield_Qty { get; set; }
        public decimal? IN_Qty { get; set; }
        public int? QTY { get; set; }
        public string Nation { get; set; }
        public string BG_Color { get => GetBGColor(Plan_Finish_Date); }
        public string Building { get; set; }
        public string PDC_ID { get; set; }
        public string BG_Color2 { get; set; }
        public string BG_Color3 { get; set; }
        public string Dept_ID { get; set; }

        private static string GetBGColor(string Plan_Finish_Date)
        {
            var date = Convert.ToDateTime(Plan_Finish_Date).Date;
            double totalDays = (date - DateTime.Now.Date).TotalDays;
            string result = totalDays >= 3 ? "green" : totalDays == 2 ? "yellow" : "red";

            return result;
        }
    }
}

File: eTierMeeting_v2/SPA/src/app/_core/_models/deliveryDTO.ts
export interface DeliveryDTO {
    comfirmed_Date: string;
    plan_Finish_Date: string;
    plan_Ship_Date: string;
    plan_Ship_Date_N: string | null;
    line_ID_ASY: string;
    line_ID_STC: string;
    mO_No: string;
    mO_Seq: string;
    model_Name: string;
    article: string;
    plan_Qty: number;
    uTN_Yield_Qty: number;
    iN_Qty: number | null;
    qty: number | null;
    nation: string;
    bG_Color: string;
    building: string;
    pDC_ID: string;
    bG_Color2: string;
    bG_Color3: string;
}
